Tests/NIOHTTP1Tests/HTTPServerUpgradeTests.swift

```diff
@@ -866,14 +866,7 @@ class HTTPServerUpgradeTestCase: XCTestCase {
 
         let channel = EmbeddedChannel()
         defer {
-            do {
-                let complete = try channel.finish()
-                XCTAssertTrue(complete.isClean)
-            } catch No.no {
-                // ok
-            } catch {
-                XCTFail("Unexpected error: \(error)")
-            }
```

Comment:
This is not a semantic replacement for the above. This used to tolerate an error that now is not tolerated.

Reply:
@Lukasa indeed. That's what I meant with "tighten" in the description. It's on EmbeddedChan so it should be deterministic and the tests pass, shouldn't we take this?